src/System.Management.Automation/engine/CommandDiscovery.cs

```diff
@@ -1328,6 +1328,15 @@ internal LookupPathCollection GetLookupDirectoryPaths()
                     foreach (string directory in tokenizedPath)
                     {
                         string tempDir = directory.TrimStart();
+                        if (tempDir.EqualsOrdinalIgnoreCase("~"))
```

Comment:
This should only check the very first directory, right? I don't know if tilda can be in the middle...

Reply:
I'd expect that we expand tilde for all elements. There are other applications which add their paths.

Reply:
My intent here is to only handle the beginning case which is the most common case.  `~` in the middle doesn't make sense.